Authentication

elijahverdoorn opened the pull request:
Add login with google button, to use google's sign-in.

Developed in parallel with https://github.com/oleville/oleville-voting/pull/22, which adds the server-side of this work.

elijahverdoorn committed:
Add temp files to gitignore

elijahverdoorn committed:
Add authentication and login with google button

rye commented on src/lib/auth.js:
```diff
@@ -0,0 +1,8 @@
+export default function postTokenToServer(tok) {
+	fetch('http://localhost:4000/login/google', {
```
I'd like to set up a configuration for this, but this works.

elijahverdoorn replied:
Yeah, a config file in the client will be good.

rye commented on src/home/index.js:
```diff
@@ -1,9 +1,34 @@
+					<p> Welcome to Oleville Voting! </p>
+					<p> Please log in to access your ballot.</p>
+					<GoogleLogin
+						clientId='437117106108-miaqebkeantfo00063nottgvsd419ns9.apps.googleusercontent.com'
```
Should we publish this?

elijahverdoorn replied:
We have to. It's not meant to be secret, there's a separate _clientSecret_ for that. This is to ensure that the token that the user sends us is a token meant for our app. When we do the verification step with google, they send us the clientId that generated the token and we check it against our clientId. If they don't match, then the login process was tampered with and we reject the token, preventing issues with this going out to the user.

rye reviewed (commented):
Looks good, but CI is red for some reason.

elijahverdoorn commented:
Alright - this PR is getting stale and we need to make a decision. As I see it, there are 2 options here:
* use the component seen here for google sign-in, accepting that we can't fully test the area that it is used.
* write a new component for google sign-in (I can't find another that does what we want)

Thoughts, @erichkauffman, @rye?

erichkauffman commented:
I think that if it does what we want it to do, we should forgo testing. With that, I think we need to keep in mind that if the area can't be tested then we should be cautious when adding new items to the area, especially items that really need testing.

rye commented:
I would tend towards the first option, accepting the testing problems.  We should poke at that some more and see if we can fix the errors in our testing situation.

elijahverdoorn committed:
Remove failing test

After discussion, we decided that the failing test should be removed for
now and we should focus on getting a working product before we worry
about that specific test.

elijahverdoorn commented:
Alright - failing test removed. We're officially running without tests, but we're closer to a working flow. Can I get an approval, @erichkauffman or @rye?
